hvac/api/auth_methods/approle.py

```diff
@@ -237,6 +237,8 @@ def generate_secret_id(
         metadata=None,
         cidr_list=None,
         token_bound_cidrs=None,
+        num_uses=None,
+        ttl=None,
```

Comment:
these need to be at the end of the parameter list in case of clients using position parameters

Reply:
@briantist fixed positional arguments and added some tests. Integration tests are the only thing actually verifying something but I still added the new arguments to the unit tests as well...